src/widgets/Notebook.cpp

```diff
@@ -165,6 +178,31 @@ void Notebook::removePage(QWidget *page)
     this->performLayout(true);
 }
 
+void Notebook::duplicatePage(QWidget *page)
+{
+    auto *item = this->findItem(page);
+    assert(item != nullptr);
+
+    auto *container = dynamic_cast<SplitContainer *>(item->page);
+    if (!container)
+    {
+        return;
+    }
+
+    auto descriptor = container->buildDescriptor();
+
+    auto *newContainer = new SplitContainer(this);
+    newContainer->applyFromDescriptor(descriptor);
+
+    int newTabPosition = this->indexOf(page) + 1;
+    auto *tab = this->addPageAt(
+        newContainer, newTabPosition,
+        item->tab->hasCustomTitle() ? item->tab->getCustomTitle() : "", false);
+    tab->setHighlightState(item->tab->highlightState());
```

Comment:
> This currently crashes if you try to duplicate a previously duplicated tab - something with attempting to access the clicked item's highlight state - `item->tab` seems to be a nullptr/invalid ptr

`item` is invalid.

---

This accesses `item` after it's invalidated due to a resize of `items_` by `addPageAt` (`item` points to a location inside that list). The following fixes the crash:

```diff
diff --git a/src/widgets/Notebook.cpp b/src/widgets/Notebook.cpp
index 0350a1d7..4bbc6148 100644
--- a/src/widgets/Notebook.cpp
+++ b/src/widgets/Notebook.cpp
@@ -195,10 +195,11 @@ void Notebook::duplicatePage(QWidget *page)
     newContainer->applyFromDescriptor(descriptor);
 
     int newTabPosition = this->indexOf(page) + 1;
+    auto highlightState = item->tab->highlightState();
     auto *tab = this->addPageAt(
         newContainer, newTabPosition,
         item->tab->hasCustomTitle() ? item->tab->getCustomTitle() : "", false);
-    tab->setHighlightState(item->tab->highlightState());
+    tab->setHighlightState(highlightState);
 
     newContainer->setTab(tab);
 }
```

Reply:
I knew I would be missing something, thanks.